tests/file_system_cache_test.py

```diff
@@ -56,5 +57,63 @@ def test_empty_key_get(self):
      with self.assertRaisesRegex(ValueError , r"key cannot be empty"):
        cache.get("")
 
+  def test_size_of_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      cache.put("foo", b"bar")
+      self.assertEqual(cache._get_cache_directory_size(), 3)
+
+  def test_size_of_empty_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      self.assertEqual(cache._get_cache_directory_size(), 0)
+
+  def test_size_of_existing_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache1 = FileSystemCache(tmpdir)
+      cache1.put("foo", b"bar")
+      del cache1
+      cache2 = FileSystemCache(tmpdir)
+      self.assertEqual(cache2._get_cache_directory_size(), 3)
+
+  def test_cache_is_full(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"the")
+      self.assertEqual(cache.get("foo"), None)
+      self.assertEqual(cache.get("second"), b"two")
+      self.assertEqual(cache.get("third"), b"the")
+
+  def test_delete_multiple_files(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"three")
+      self.assertEqual(cache.get("foo"), None)
+      self.assertEqual(cache.get("second"), None)
+      self.assertEqual(cache.get("third"), b"three")
+
+  def test_least_recently_accessed_file(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      #Added sleeps because these operations were executing too fast that the access time could not be captured properly
```

Comment:
Minor nits on verb tense and line length, please change to:
```python
      # Sleep because otherwise these operations execute too fast and
      # the access time isn't captured properly.
```